ngrep.c

```diff
@@ -1557,4 +1573,27 @@ char *win32_choosedevice(void) {
 }
 #endif
 
+/* Parse an unsigned long, exit program if anything goes wrong. */
+unsigned long parseulong(const char *s) {
+    unsigned long n = 0;
+    char *end = NULL;
 
+    errno = 0;
+    n = strtoul(s, &end, 0);
+
+    if (errno != 0 || *s == '\0' || end == NULL || *end != '\0') {
+        if (errno != 0) { /* GNU strtoul will set errno. */
+            perror("strtoul");
+        }
+        fprintf(stderr, "Could not convert unsigned long: `%s'.\n", s);
+
+        clean_exit(-1);
+    }
+
+    return n;
+}
+
+void dierange(char *msg, unsigned long value) {
+    fprintf(stderr, "`%s': value `%lu' is out of range.\n", msg, value);
```

Comment:
If this routine expected a character array as the second parameter I think the feedback would be more valuable since in some cases we won't be able to output `value` properly as an `unsigned long`.